gamemodes/terrortown/entities/entities/ttt_game_text.lua

```diff
@@ -54,28 +54,23 @@ end
 function ENT:AcceptInput(name, activator)
 	if name == "Display" then
 		local recv = activator
-
-		local r = self.Receiver
-		if r == RECEIVE_ALL then
-			recv = nil
-		elseif r == RECEIVE_DETECTIVE then
-			recv = GetRoleChatFilter(ROLE_DETECTIVE)
-		elseif r == RECEIVE_TRAITOR then
-			recv = GetTeamChatFilter(TEAM_TRAITOR)
-		elseif r == RECEIVE_INNOCENT then
-			recv = GetTeamChatFilter(TEAM_INNOCENT)
-		elseif r == RECEIVE_ACTIVATOR then
-			if not IsValid(activator) or not activator:IsPlayer() then
-				ErrorNoHalt("ttt_game_text tried to show message to invalid !activator\n")
-
-				return true
-			end
-		elseif r == RECEIVE_CUSTOMROLE and self.teamReceiver then
-			recv = GetTeamChatFilter(self.teamReceiver)
+				
+		if IsValid(activator) and activator:IsPlayer() then
+			local receiver_tbl = { -- i think this is faster than chaining if/elseif
+				RECEIVE_ALL = nil,
+				RECEIVE_DETECTIVE = GetRoleChatFilter(ROLE_DETECTIVE),
+				RECEIVE_TRAITOR = GetRoleChatFilter(TEAM_TRAITOR),
+				RECEIVE_INNOCENT = GetRoleChatFilter(TEAM_INNOCENT)
+			}
+			
+			recv = (self.teamReceiver) ? GetTeamChatFilter(self.teamReceiver) : receiver_tbl[self.Receiver]
+			CustomMsg(recv, self.Message, self.Color)
+			return true -- success
+		else
+			ErrorNoHalt("ttt_game_text tried to show message to invalid !activator\n")
+			return false -- failed, either invalid activator or activator was not a player
```

Comment:
Why did you change the return value? Not saying that this is wrong, but i'd like an explanation for why you return false here when the old code would have returned true instead.

A small nitpick: Please remove the comment, it doesn't add value imo

Reply:
i dont understand returning True when an error occurred and the script failed due to invalid activator

is there some trick somewhere else that relies on this one being True?